R/imputation.R

```diff
@@ -403,8 +510,44 @@ impute_with <- function(x, val) {
 ##' }
 ##'
 ##' impute_matrix(m, FUN = random_imp)
-impute_fun <- function(x, FUN, ...) {
+impute_fun <- function(x, FUN, margin = 1L, ...) {
+    margin <- .checkMargin(margin)
+    if (margin == 2L) x <- t(x)
     res <- do.call(FUN, list(x, ...))
+    if (margin == 2L) res <- t(res)
     stopifnot(identical(dim(x), dim(res)))
     res
-}
\ No newline at end of file
+}
+
+.checkMargin <- function(margin) {
+    margin <- as.integer(margin)
+    if (!margin %in% c(1L, 2L))
+        stop("'margin' must be 1L or 2L")
+    message("Imputing along margin ", margin, ".")
```

Comment:
Maybe explicit what margin 1 and 2 mean? eg

```r
message("Imputing along margin ", margin, ifelse(margin == 1L, " (features, rows)", " (samples, columns)"), ".")
```

Reply:
OK